src/main/java/com/mraof/minestuck/network/EditmodeDragPacket.java

```diff
@@ -0,0 +1,342 @@
+package com.mraof.minestuck.network;
+
+import com.mraof.minestuck.MinestuckConfig;
+import com.mraof.minestuck.alchemy.*;
+import com.mraof.minestuck.computer.editmode.*;
+import com.mraof.minestuck.player.PlayerSavedData;
+import com.mraof.minestuck.skaianet.SburbConnection;
+import com.mraof.minestuck.util.MSCapabilities;
+import com.mraof.minestuck.util.MSSoundEvents;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.BlockItem;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.context.UseOnContext;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.SoundType;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.gameevent.GameEvent;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.Vec3;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public final class EditmodeDragPacket
+{
+	private static final Logger LOGGER = LogManager.getLogger();
+	
+	private static boolean editModePlaceCheck(Level level, Player player, InteractionHand hand)
+	{
+		if(!level.isClientSide() && ServerEditHandler.getData(player) != null)
```

Comment:
These should already be checked by the caller of this function, so I recommend taking in the EditData as a parameter instead.